src/main/java/org/sonar/plugins/groovy/surefire/GroovySurefireParser.java

```diff
@@ -0,0 +1,201 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonar.plugins.groovy.surefire;
+
+import com.google.common.collect.Lists;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonar.api.BatchExtension;
+import org.sonar.api.batch.SensorContext;
+import org.sonar.api.batch.fs.FilePredicate;
+import org.sonar.api.batch.fs.FilePredicates;
+import org.sonar.api.batch.fs.FileSystem;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.component.ResourcePerspectives;
+import org.sonar.api.measures.CoreMetrics;
+import org.sonar.api.measures.Metric;
+import org.sonar.api.test.MutableTestPlan;
+import org.sonar.api.test.TestCase;
+import org.sonar.api.utils.ParsingUtils;
+import org.sonar.api.utils.SonarException;
+import org.sonar.api.utils.StaxParser;
+import org.sonar.plugins.groovy.foundation.Groovy;
+import org.sonar.plugins.groovy.surefire.data.SurefireStaxHandler;
+import org.sonar.plugins.groovy.surefire.data.UnitTestClassReport;
+import org.sonar.plugins.groovy.surefire.data.UnitTestIndex;
+import org.sonar.plugins.groovy.surefire.data.UnitTestResult;
+
+import javax.xml.stream.XMLStreamException;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.util.List;
+import java.util.Map;
+
+public class GroovySurefireParser implements BatchExtension {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(GroovySurefireParser.class);
+  private final Groovy groovy;
+  private final ResourcePerspectives perspectives;
+  private final FileSystem fs;
+
+  public GroovySurefireParser(Groovy groovy, ResourcePerspectives perspectives, FileSystem fs) {
+    this.groovy = groovy;
+    this.perspectives = perspectives;
+    this.fs = fs;
+  }
+
+
+  public void collect(SensorContext context, File reportsDir) {
+    File[] xmlFiles = getReports(reportsDir);
+    if (xmlFiles.length > 0) {
+      parseFiles(context, xmlFiles);
+    }
+  }
+
+  private File[] getReports(File dir) {
+    if (dir == null) {
+      return new File[0];
+    } else if (!dir.isDirectory()) {
+      LOGGER.warn("Reports path not found: " + dir.getAbsolutePath());
+      return new File[0];
+    }
+    File[] unitTestResultFiles = findXMLFilesStartingWith(dir, "TEST-");
+    if (unitTestResultFiles.length == 0) {
+      // maybe there's only a test suite result file
+      unitTestResultFiles = findXMLFilesStartingWith(dir, "TESTS-");
+    }
+    return unitTestResultFiles;
+  }
+
+  private File[] findXMLFilesStartingWith(File dir, final String fileNameStart) {
+    return dir.listFiles(new FilenameFilter() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this a named "static" inner class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2694)